### PR TITLE
Add keycloak integration

### LINKED ISSUE #34: [RFE] Add SSO logic
We didn't implement yet the logic to configure SSO like in `pulp-operator`.

### PR DESCRIPTION
**Pull request by git-hyagi**
closes #34

### CODE REVIEW

File: controllers/sso_configuration.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	requiredKeys := []string{
+		"social_auth_keycloak_key", "social_auth_keycloak_secret", "social_auth_keycloak_public_key",
+		"keycloak_host", "keycloak_protocol", "keycloak_port", "keycloak_realm",
+	}
+
+	optionalKeys := []string{
+		"keycloak_admin_role", "keycloak_group_token_claim", "keycloak_role_token_claim", "keycloak_host_loopback",
+	}
```
Comment (fao89) [inline controllers/sso_configuration.go:38]: should this optional x required logic be moved to: [retrieveSecretData](https://github.com/git-hyagi/pulp-operator-go/blob/main/controllers/common.go#L37) ?

File: controllers/sso_configuration.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+func (r *PulpReconciler) ssoConfig(ctx context.Context, pulp *repomanagerv1alpha1.Pulp, pulpSettings *string) error {
+	log := ctrllog.FromContext(ctx)
+
+	// Check for specified sso configuration secret
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, types.NamespacedName{Name: pulp.Spec.SSOSecret, Namespace: pulp.Namespace}, secret)
+	if err != nil {
+		log.Error(err, "Failed to find "+pulp.Spec.SSOSecret+" secret")
+		return err
+	}
```
Comment (fao89) [inline controllers/sso_configuration.go:24]: why not use [retrieveSecretData](https://github.com/git-hyagi/pulp-operator-go/blob/main/controllers/common.go#L37) ?

**Review by fao89 — APPROVED** [review]
LGTM!
Just left a couple of suggestions

File: controllers/sso_configuration.go (hunk 2 above)
Comment (git-hyagi) [reply controllers/sso_configuration.go:24]: Good idea!
Working on it

File: controllers/common.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func createPwd(pwdSize int) string {
 }
 
 // Retrieve specific keys from secret object
-func (r *PulpReconciler) retrieveSecretData(ctx context.Context, secretName, secretNamespace string, keys ...string) (map[string]string, error) {
+func (r *PulpReconciler) retrieveSecretData(ctx context.Context, secretName, secretNamespace string, required bool, keys ...string) (map[string]string, error) {
```
Comment (fao89) [inline controllers/common.go:37]: I don't know how golang works, can we make required=True as default?

**Review by fao89 — APPROVED** [review]

File: controllers/common.go (hunk 3 above)
Comment (fao89) [reply controllers/common.go:37]: actually it isn't possible: https://github.com/golang/go/issues/1675

**Pull request merged**